src/execution_plan/ops/op_procedure_call.c

```diff
@@ -0,0 +1,119 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "op_procedure_call.h"
+#include "../../util/arr.h"
+
+static void _yield(OpProcCall *op, SIValue *proc_output, Record r) {
+    // TODO: map AST_GetAliasID values.
+    if (!op->yield_map) {
+        op->yield_map = rm_malloc(sizeof(OutputMap) * array_len(op->output));
+
+        for(uint i = 0; i < array_len(op->output); i++) {
+            char *yield = op->output[i];
+            for(uint j = 0; j < array_len(proc_output); j+=2) {
+                char *key = (proc_output + j)->stringval;
+                SIValue *val = proc_output + (j+1);
```

Comment:
I think constructions like this are a bit more readable:
`SIValue *val = &proc_output[j+1];`

Reply:
👍